ImageViewer.php

```diff
@@ -250,12 +285,14 @@ function renderPreview($instrument, $record, $event_id) {
         }
 
         // We need to know the filetype to validate when the file has been previously uploaded...
-        $q = REDCap::getData('json',$record, array_keys($fields), $event_id);
-        $results = json_decode($q, true);
-        $result = $results[0];
+        /*$q = REDCap::getData('json',$record, array_keys($fields), $event_id);
```

Comment:
@jwalsoe - is it not possible to obtain the doc_id from the standard getData by parsing out the repeat instance value?  Just curious as to why you had to add the sql lookup function.

Reply:
Maybe it is possible, I just looked at the getData documentation and saw that it had no parameter for repeat instance, and I was in a hurry (one project needed this asap). For me -this time- using the sql lookup was the quickest way to get this done. I have much better tools for working with the database than for working towards the hook/plugin functions. It probably is a better idea to rewrite this stuff to just use the getData output. I might venture to do this once I have better time, provided no one else does it before me :-)